Sources/AWSLambdaRuntimeCore/LambdaContext.swift

```diff
@@ -52,6 +53,9 @@ extension Lambda {
     /// Lambda runtime context.
     /// The Lambda runtime generates and passes the `Context` to the Lambda handler as an argument.
     public final class Context: CustomDebugStringConvertible {
+        // TODO: use RWLock (separate PR)
+        private let lock = Lock()
```

Comment:
tbh we can start out with just a lock and change only if proven to matter a lot; 

The https://blog.nelhage.com/post/rwlock-contention/ keeps being brought up when we reach for RWLocks recently

Reply:
RWlock are great when you do single (or very very very little) write and all-reads. in mixed mode it can get tricky tp get good performance